TT-13174 Operator and Sync and Charts release (#5460)
TykTechnologies/tyk-docs: tyk-docs/content/basic-config-and-security/security/mutual-tls/client-mtls.md

Here's what it should look like:
{{< img src="/img/2.10/client_mtls_multiple_auth.png" alt="enable_cert" >}}

> ### Define via Tyk Operator (Tyk Classic){#tyk-operator-classic}

Reviewer: What does it mean "Tyk Classic"?

> ### Define via Tyk Operator (Tyk Classic){#tyk-operator-classic}

Reviewer: ```suggestion
### Setup via Tyk Operator (Tyk Classic){#tyk-operator-classic}
```

1. **Generate Self-Signed Key Pair:**

You can generate a self-signed key pair using the following OpenSSL command:

```bash
openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -nodes
```

2. **Create Kubernetes Secret:**

Create a secret in Kubernetes to store the client certificate:

```bash
kubectl create secret tls my-test-tls --cert cert.pem --key key.pem
```

3. **Create API Definition:**

Below is the YAML configuration for an API that uses mTLS authentication. Note that the `client_certificate_refs` field references the Kubernetes secret created in the previous step.

```yaml {hl_lines=["19-21"],linenos=false}
apiVersion: tyk.tyk.io/v1alpha1
kind: ApiDefinition
metadata:
  name: httpbin-client-mtls
spec:
  name: Httpbin Client MTLS
  protocol: http
  active: true
  proxy:
    target_url: http://httpbin.org
    listen_path: /httpbin
    strip_listen_path: true
  version_data:
    default_version: Default
    not_versioned: true
    versions:
      Default:
        name: Default
  use_mutual_tls_auth: true
  client_certificate_refs:
    - my-test-tls
```

> ### Define via Tyk Operator (Tyk OAS){#tyk-operator-oas}

Reviewer: ```suggestion
### Setup Static mTLS with Tyk Operator (Tyk OAS API Definition){#tyk-operator-oas}
```

> For Tyk OAS APIs, client certificates are managed using the `TykOasApiDefinition` CRD. You can reference Kubernetes secrets that store client certificates in your API definitions.

Reviewer: ```suggestion
In Tyk OAS API Definition, client certificates are managed using the `TykOasApiDefinition` CRD. You can reference Kubernetes secrets that store client certificates in your API definitions.
```

Example of Referencing Client Certificates in Tyk OAS

```yaml {hl_lines=["48-50"],linenos=false}
# Secret is not created in this manifest.
# Please store client certificate in k8s TLS secret `tls-cert`.

apiVersion: v1
data:
  test_oas.json: |-
    {
        "info": {
          "title": "Petstore",
          "version": "1.0.0"
        },
        "openapi": "3.0.3",
        "components": {},
        "paths": {},
        "x-tyk-api-gateway": {
          "info": {
            "name": "Petstore",
            "state": {
              "active": true
            }
          },
          "upstream": {
            "url": "https://petstore.swagger.io/v2"
          },
          "server": {
            "listenPath": {
              "value": "/petstore/",
              "strip": true
            }
          }
        }
      }
kind: ConfigMap
metadata:
  name: cm
  namespace: default
---
apiVersion: tyk.tyk.io/v1alpha1
kind: TykOasApiDefinition
metadata:
  name: petstore
spec:
  tykOAS:
    configmapRef:
      name: cm
      namespace: default
      keyName: test_oas.json
  clientCertificate: 
      enabled: true
      allowlist: [tls-cert]
```

> In this example, the `clientCertificate` section allows you to enable client certificate management and specify a list of Kubernetes secrets (`tls-cert`) that store allowed client certificates.

Reviewer: @caroltyk in my opinion, since the code snippet is long, it's better to explain the example beforehand, so users don't have to search on their own what it means. Moving the text before the code should sort it this ux

Author: updated
